Assignment4

AnnaaKorzhenko opened the pull request:
implemented few sproc (IN, OUT and INOUT) and used transaction with a commit logic

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+
+CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
+BEGIN
+    SELECT * FROM teams WHERE team_name = teamName;
```
Do not use * in production code

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+-- creating a procedure
+DELIMITER &&
+
+CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
```
Rename to get_team_info_sproc

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+-- it gives total count of existing players as an output
+DELIMITER &&
+
+CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
```
Rename to get_player_count_sproc

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+
+CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
+BEGIN
+SELECT COUNT(*) INTO totalPlayers FROM players;
```
COUNT(1)

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+END&&
+DELIMITER ;
+   
+SET @totalPLayers = NULL;
```
I guess it is a duplicate from the executions file. If yes then remove it

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+-- it takes player id as an IN parameter and then returns number of goals scored bt this player as an OUT parameter, but using INOUT
+DELIMITER &&
+
+CREATE PROCEDURE CalculatePlayerGoals (
```
rename to calculate_player_goals_sproc

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+    INOUT playerID INT
+)
+BEGIN
+    DECLARE totalGoals INT;
```
rename to total_goals

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+)
+BEGIN
+    DECLARE totalGoals INT;
+SELECT COUNT(*) INTO totalGoals FROM goals WHERE player_id = playerID;
```
COUNT(1)

Ivashchyk commented on pa4procedures.sql:
```diff
@@ -0,0 +1,75 @@
+------------- TRANSACTION -------------------------
+-- if the input data about match's goals is accurate (not negative), it updates the info in "matches" table and commits
+-- else, it still updates, but after the "if" condition, it does rollback
+CREATE PROCEDURE update_match_result(
```
rename to update_match_result_sproc

Ivashchyk reviewed (changes requested):
minor issues

AnnaaKorzhenko commented:
Hello! I have updated my assignment due to your recommendations and
requested changes. I changed the naming and avoided using *
Here's the needed commit in the PR:
https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3/commits/6a098597988ba1a600e7f869de3ff500498e6bcb

On Mon, Apr 8, 2024 at 9:11 PM Anton Ivashchyk ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> minor issues
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556232327>
> :
>
> > @@ -0,0 +1,75 @@
> +----------- IN -----------------------
> +-- it takes a team name as an IN parameter and selects all the information about that exact team
> +-- creating a procedure
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
> +BEGIN
> +    SELECT * FROM teams WHERE team_name = teamName;
>
> Do not use * in production code
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556233376>
> :
>
> > @@ -0,0 +1,75 @@
> +----------- IN -----------------------
> +-- it takes a team name as an IN parameter and selects all the information about that exact team
> +-- creating a procedure
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
>
> Rename to get_team_info_sproc
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556234349>
> :
>
> > +-- creating a procedure
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
> +BEGIN
> +    SELECT * FROM teams WHERE team_name = teamName;
> +END&&
> +
> +DELIMITER ;
> +
> +
> +----------- OUT ------------------------
> +-- it gives total count of existing players as an output
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
>
> Rename to get_player_count_sproc
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556235118>
> :
>
> > +
> +CREATE PROCEDURE GetTeamInfo (IN teamName VARCHAR(255))
> +BEGIN
> +    SELECT * FROM teams WHERE team_name = teamName;
> +END&&
> +
> +DELIMITER ;
> +
> +
> +----------- OUT ------------------------
> +-- it gives total count of existing players as an output
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
> +BEGIN
> +SELECT COUNT(*) INTO totalPlayers FROM players;
>
> COUNT(1)
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556237087>
> :
>
> > +END&&
> +
> +DELIMITER ;
> +
> +
> +----------- OUT ------------------------
> +-- it gives total count of existing players as an output
> +DELIMITER &&
> +
> +CREATE PROCEDURE GetPlayerCount (OUT totalPlayers INT)
> +BEGIN
> +SELECT COUNT(*) INTO totalPlayers FROM players;
> +END&&
> +DELIMITER ;
> +
> +SET @totalPLayers = NULL;
>
> I guess it is a duplicate from the executions file. If yes then remove it
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556238378>
> :
>
> > +BEGIN
> +SELECT COUNT(*) INTO totalPlayers FROM players;
> +END&&
> +DELIMITER ;
> +
> +SET @totalPLayers = NULL;
> +CALL ***@***.***);
> +SELECT @totalPlayers;
> +
> +
> +
> +---------- INOUT ------------------------
> +-- it takes player id as an IN parameter and then returns number of goals scored bt this player as an OUT parameter, but using INOUT
> +DELIMITER &&
> +
> +CREATE PROCEDURE CalculatePlayerGoals (
>
> rename to calculate_player_goals_sproc
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556238622>
> :
>
> > +
> +SET @totalPLayers = NULL;
> +CALL ***@***.***);
> +SELECT @totalPlayers;
> +
> +
> +
> +---------- INOUT ------------------------
> +-- it takes player id as an IN parameter and then returns number of goals scored bt this player as an OUT parameter, but using INOUT
> +DELIMITER &&
> +
> +CREATE PROCEDURE CalculatePlayerGoals (
> +    INOUT playerID INT
> +)
> +BEGIN
> +    DECLARE totalGoals INT;
>
> rename to total_goals
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556238772>
> :
>
> > +SET @totalPLayers = NULL;
> +CALL ***@***.***);
> +SELECT @totalPlayers;
> +
> +
> +
> +---------- INOUT ------------------------
> +-- it takes player id as an IN parameter and then returns number of goals scored bt this player as an OUT parameter, but using INOUT
> +DELIMITER &&
> +
> +CREATE PROCEDURE CalculatePlayerGoals (
> +    INOUT playerID INT
> +)
> +BEGIN
> +    DECLARE totalGoals INT;
> +SELECT COUNT(*) INTO totalGoals FROM goals WHERE player_id = playerID;
>
> COUNT(1)
> ------------------------------
>
> In pa4procedures.sql
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#discussion_r1556239059>
> :
>
> > +    DECLARE totalGoals INT;
> +SELECT COUNT(*) INTO totalGoals FROM goals WHERE player_id = playerID;
> +SET playerID = totalGoals;
> +END&&
> +
> +DELIMITER ;
> +
> +SET @param = 5;
> +CALL ***@***.***);
> +SELECT @param;
> +
> +
> +------------- TRANSACTION -------------------------
> +-- if the input data about match's goals is accurate (not negative), it updates the info in "matches" table and commits
> +-- else, it still updates, but after the "if" condition, it does rollback
> +CREATE PROCEDURE update_match_result(
>
> rename to update_match_result_sproc
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3#pullrequestreview-1987101129>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD3AGJ2GQJRQYURUIWCEBVDY4LMUVAVCNFSM6AAAAABFWAAARSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSOBXGEYDCMJSHE>
> .
> You are receiving this because you authored the thread.Message ID:
> <x01-software-engineering/dbe-assignment-01-AnnaaKorzhenko/pull/3/review/1987101129
> @github.com>
>
